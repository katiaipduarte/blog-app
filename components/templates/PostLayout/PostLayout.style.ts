import styled from 'styled-components';

export const PostContainer = styled.article`
  padding: 0 1rem;
  max-width: 48rem;
  margin: 0 auto;
  text-align: center;

  header {
    padding-top: 1.5rem;

    dl {
      dd {
        line-height: 1.5rem;
        font-size: 1rem;
        color: ${({ theme }) => theme.grey};
      }
    }

    h1 {
      line-height: 2.25rem;
      font-size: 1.875rem;
      font-weight: 800;
    }
  }

  .post {
    padding-bottom: 2rem;
    .author {
      padding: 2rem 0;
      border-bottom: 1px solid ${({ theme }) => theme.text};
      dd {
        font-size: 0.875rem;
        font-weight: 500;
        white-space: nowrap;
        line-height: 1.25rem;
      }
    }

    .content {
      padding: 2rem 0;
      text-align: justify;
      border-bottom: 1px solid ${({ theme }) => theme.text};

      p {
        margin: 1.25em 0;
        font-size: 1rem;
        line-height: 1.75;
      }
    }

    footer {
      text-align: justify;
      padding-top: 2rem;

      .other-articles {
        display: flex;
        justify-content: space-between;
        .previous-article,
        .next-article {
          padding-bottom: 1rem;
          h2 {
            font-size: 0.75rem;
            line-height: 1rem;
            color: ${({ theme }) => theme.grey};
            text-transform: uppercase;
          }
          button {
            padding-top: 0rem;
            text-transform: capitalize;
          }
        }
      }

      .go-back {
        padding-top: 1rem;
        float: right;
      }
    }
  }
`;
