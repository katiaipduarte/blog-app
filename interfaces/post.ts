export type Post = {
  author: string;
  content: string;
  description: string;
  id: number;
  publish_date: string;
  slug: string;
  title: string;
};
