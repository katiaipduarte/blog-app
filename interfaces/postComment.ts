export type PostComment = {
  content: string;
  date: string;
  id: number;
  parent_id: number | null;
  postId: number;
  user: string;
};
