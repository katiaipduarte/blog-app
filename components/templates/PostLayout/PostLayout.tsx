import { Post } from '@interfaces/post';
import React from 'react';
import { PostContainer } from './PostLayout.style';
import Link from 'next/link';

type Props = {
  post: Post;
  comments: Comment[];
};

const PostLayout = (props: Props): JSX.Element => {
  const { post, comments } = props;

  const date = new Date(post.publish_date).toLocaleDateString('en', {
    weekday: 'long',
    year: 'numeric',
    month: 'long',
    day: 'numeric'
  });

  return (
    <PostContainer>
      <header>
        <dl>
          <dd>
            <time
              aria-label={`Published on ${date}`}
              dateTime={post.publish_date}
              tabIndex={0}
            >
              {date}
            </time>
          </dd>
        </dl>
        <h1>{post.title}</h1>
      </header>
      <div className="post" style={{ gridTemplateRows: 'auto 1fr' }}>
        <dl
          className="author"
          aria-label={`Post written by ${post.author}`}
          tabIndex={0}
        >
          <dd>{post.author}</dd>
        </dl>
        <div
          className="content"
          dangerouslySetInnerHTML={{ __html: post.content }}
          tabIndex={0}
          role="text"
        ></div>
        <footer>
          <Link href={`/blog`} aria-label={`Go back to the blog`}>
            <button className="transparent" role="button">
              &larr; Back to the blog
            </button>
          </Link>
        </footer>
      </div>
    </PostContainer>
  );
};

export default PostLayout;
