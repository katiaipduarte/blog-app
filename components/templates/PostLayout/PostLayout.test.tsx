import { Post } from '@interfaces/post';
import { PostComment } from '@interfaces/postComment';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import PostLayout from './PostLayout';

const mockedPost: Post = {
  author: 'author',
  content: 'bla',
  description: 'test',
  id: 1,
  publish_date: '2021-01-01',
  slug: 'test',
  title: 'test'
};

const mockedComments: PostComment[] = [
  {
    content: 'banana',
    user: 'test',
    date: '2021-06-28',
    parent_id: null,
    postId: 10,
    id: 8
  },
  {
    content: 'bananan 2x',
    user: 'yolo',
    date: '2021-06-28',
    parent_id: 8,
    postId: 10,
    id: 9
  }
];

describe('the <PostLayout /> component', () => {
  it('should render Post empty', () => {
    const tree = renderer
      .create(<PostLayout post={{} as Post} comments={[] as PostComment[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Post with 0 comments', () => {
    const tree = renderer
      .create(<PostLayout post={mockedPost} comments={[] as PostComment[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Post with comments', () => {
    const tree = renderer
      .create(<PostLayout post={mockedPost} comments={mockedComments} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
