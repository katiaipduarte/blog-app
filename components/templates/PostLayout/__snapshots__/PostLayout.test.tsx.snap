// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <PostLayout /> component should render Post empty 1`] = `
.c0 {
  padding: 0 1rem;
  max-width: 48rem;
  margin: 0 auto;
  text-align: center;
}

.c0 header {
  padding-top: 1.5rem;
}

.c0 header dl dd {
  line-height: 1.5rem;
  font-size: 1rem;
}

.c0 header h1 {
  line-height: 2.25rem;
  font-size: 1.875rem;
  font-weight: 800;
}

.c0 .post {
  padding-bottom: 2rem;
}

.c0 .post .author {
  padding: 2rem 0;
  border-bottom: 1px solid;
}

.c0 .post .author dd {
  font-size: 0.875rem;
  font-weight: 500;
  white-space: nowrap;
  line-height: 1.25rem;
}

.c0 .post .content {
  padding: 2rem 0;
  text-align: justify;
  border-bottom: 1px solid;
}

.c0 .post .content p {
  margin: 1.25em 0;
  font-size: 1rem;
  line-height: 1.75;
}

.c0 .post footer {
  text-align: justify;
  padding-top: 2rem;
}

.c0 .post footer .other-articles {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .post footer .other-articles .previous-article,
.c0 .post footer .other-articles .next-article {
  padding-bottom: 1rem;
}

.c0 .post footer .other-articles .previous-article h2,
.c0 .post footer .other-articles .next-article h2 {
  font-size: 0.75rem;
  line-height: 1rem;
  text-transform: uppercase;
}

.c0 .post footer .other-articles .previous-article button,
.c0 .post footer .other-articles .next-article button {
  padding-top: 0rem;
  text-transform: capitalize;
}

.c0 .post footer .go-back {
  padding-top: 1rem;
  float: right;
}

.c2 {
  margin: 2rem auto;
}

.c2 p {
  font-size: 1.25rem;
  line-height: 1.75rem;
}

.c2 div {
  margin: 2rem 0;
}

.c2 div input[type='text'],
.c2 div textarea {
  width: 100% !important;
  border-radius: 0.4rem;
  box-sizing: border-box;
  margin: 1rem 0;
  box-shadow: 0 0 20px 0 rgba(0,0,0,0.2),0 5px 5px 0 rgba(0,0,0,0.24);
  padding: 0.5rem;
  display: block;
}

.c2 div input[type='text']:focus-visible,
.c2 div textarea:focus-visible,
.c2 div input[type='text']:focus:not(:focus-visible),
.c2 div textarea:focus:not(:focus-visible) {
  outline: none;
  border: 2px solid;
}

.c2 div input[type='text'] {
  height: 2rem;
}

.c2 div textarea {
  height: 10rem !important;
  resize: none;
}

.c2 div .invalid {
  border: 2px solid !important;
}

.c2 div .message {
  font-size: 0.75rem;
  display: none;
}

.c2 div .show {
  display: block !important;
}

.c1 {
  margin: 2rem 0;
  text-align: justify;
}

.c1 ul {
  margin: 2rem 0;
}

.c1 ul li {
  margin: 1rem 0;
}

.c1 ul li h3 {
  font-size: 0.75rem;
}

.c1 ul li p {
  margin-top: 0.25rem;
  font-size: 0.5rem;
}

.c1 ul li .parent,
.c1 ul li .child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 ul li .child {
  width: calc(100% - 1.5rem);
  paddding: 2rem 0;
  margin: 1rem 0 1rem 1.5rem;
}

@media all and (min-width:960px) {
  .c2 {
    text-align: left;
  }

  .c2 div input[type='text'],
  .c2 div textarea {
    width: 70% !important;
  }
}

@media all and (min-width:960px) {
  .c1 ul li h3 {
    font-size: 0.8rem;
  }

  .c1 ul li p {
    margin-top: 0.25rem;
    font-size: 1rem;
  }

  .c1 ul li .child {
    max-width: 30rem;
    paddding: 2rem 0;
    margin: 1rem 0 1rem 3.5rem;
  }
}

<article
  className="c0"
>
  <header>
    <dl>
      <dt />
      <dd>
        <time
          aria-label="Published on Invalid Date"
          tabIndex={0}
        >
          Invalid Date
        </time>
      </dd>
    </dl>
    <h1 />
  </header>
  <div
    className="post"
  >
    <dl
      aria-label="Post written by undefined"
      className="author"
      tabIndex={0}
    >
      <dt />
      <dd />
    </dl>
    <div
      className="content"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
      tabIndex={0}
    />
    <section
      className="c1"
    >
      <h2
        tabIndex={0}
      >
        0 Comments
      </h2>
      <ul />
      <form
        className="c2"
        onSubmit={[Function]}
      >
        <h2>
          Submit a comment
        </h2>
        
        <div>
          <input
            className=""
            id="user"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
          <span
            className="message"
            data-testid="username-error-message"
          >
            Username cannot be empty
          </span>
        </div>
        <div>
          <textarea
            className=""
            id="content"
            name="comment"
            onChange={[Function]}
            placeholder="Write your comment here..."
            value=""
          />
          <span
            className="message"
            data-testid="comments-error-message"
          >
            Comment cannot be empty
          </span>
        </div>
        <button
          className="full"
          data-testid="submit"
          type="submit"
        >
          Submit
        </button>
      </form>
    </section>
    <footer>
      <button
        className="transparent go-back"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="button"
      >
        ‚Üê Back to the blog
      </button>
    </footer>
  </div>
</article>
`;

exports[`the <PostLayout /> component should render Post with 0 comments 1`] = `
.c0 {
  padding: 0 1rem;
  max-width: 48rem;
  margin: 0 auto;
  text-align: center;
}

.c0 header {
  padding-top: 1.5rem;
}

.c0 header dl dd {
  line-height: 1.5rem;
  font-size: 1rem;
}

.c0 header h1 {
  line-height: 2.25rem;
  font-size: 1.875rem;
  font-weight: 800;
}

.c0 .post {
  padding-bottom: 2rem;
}

.c0 .post .author {
  padding: 2rem 0;
  border-bottom: 1px solid;
}

.c0 .post .author dd {
  font-size: 0.875rem;
  font-weight: 500;
  white-space: nowrap;
  line-height: 1.25rem;
}

.c0 .post .content {
  padding: 2rem 0;
  text-align: justify;
  border-bottom: 1px solid;
}

.c0 .post .content p {
  margin: 1.25em 0;
  font-size: 1rem;
  line-height: 1.75;
}

.c0 .post footer {
  text-align: justify;
  padding-top: 2rem;
}

.c0 .post footer .other-articles {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .post footer .other-articles .previous-article,
.c0 .post footer .other-articles .next-article {
  padding-bottom: 1rem;
}

.c0 .post footer .other-articles .previous-article h2,
.c0 .post footer .other-articles .next-article h2 {
  font-size: 0.75rem;
  line-height: 1rem;
  text-transform: uppercase;
}

.c0 .post footer .other-articles .previous-article button,
.c0 .post footer .other-articles .next-article button {
  padding-top: 0rem;
  text-transform: capitalize;
}

.c0 .post footer .go-back {
  padding-top: 1rem;
  float: right;
}

.c2 {
  margin: 2rem auto;
}

.c2 p {
  font-size: 1.25rem;
  line-height: 1.75rem;
}

.c2 div {
  margin: 2rem 0;
}

.c2 div input[type='text'],
.c2 div textarea {
  width: 100% !important;
  border-radius: 0.4rem;
  box-sizing: border-box;
  margin: 1rem 0;
  box-shadow: 0 0 20px 0 rgba(0,0,0,0.2),0 5px 5px 0 rgba(0,0,0,0.24);
  padding: 0.5rem;
  display: block;
}

.c2 div input[type='text']:focus-visible,
.c2 div textarea:focus-visible,
.c2 div input[type='text']:focus:not(:focus-visible),
.c2 div textarea:focus:not(:focus-visible) {
  outline: none;
  border: 2px solid;
}

.c2 div input[type='text'] {
  height: 2rem;
}

.c2 div textarea {
  height: 10rem !important;
  resize: none;
}

.c2 div .invalid {
  border: 2px solid !important;
}

.c2 div .message {
  font-size: 0.75rem;
  display: none;
}

.c2 div .show {
  display: block !important;
}

.c1 {
  margin: 2rem 0;
  text-align: justify;
}

.c1 ul {
  margin: 2rem 0;
}

.c1 ul li {
  margin: 1rem 0;
}

.c1 ul li h3 {
  font-size: 0.75rem;
}

.c1 ul li p {
  margin-top: 0.25rem;
  font-size: 0.5rem;
}

.c1 ul li .parent,
.c1 ul li .child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 ul li .child {
  width: calc(100% - 1.5rem);
  paddding: 2rem 0;
  margin: 1rem 0 1rem 1.5rem;
}

@media all and (min-width:960px) {
  .c2 {
    text-align: left;
  }

  .c2 div input[type='text'],
  .c2 div textarea {
    width: 70% !important;
  }
}

@media all and (min-width:960px) {
  .c1 ul li h3 {
    font-size: 0.8rem;
  }

  .c1 ul li p {
    margin-top: 0.25rem;
    font-size: 1rem;
  }

  .c1 ul li .child {
    max-width: 30rem;
    paddding: 2rem 0;
    margin: 1rem 0 1rem 3.5rem;
  }
}

<article
  className="c0"
>
  <header>
    <dl>
      <dt />
      <dd>
        <time
          aria-label="Published on Friday, January 1, 2021"
          dateTime="2021-01-01"
          tabIndex={0}
        >
          Friday, January 1, 2021
        </time>
      </dd>
    </dl>
    <h1>
      test
    </h1>
  </header>
  <div
    className="post"
  >
    <dl
      aria-label="Post written by author"
      className="author"
      tabIndex={0}
    >
      <dt />
      <dd>
        author
      </dd>
    </dl>
    <div
      className="content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "bla",
        }
      }
      tabIndex={0}
    />
    <section
      className="c1"
    >
      <h2
        tabIndex={0}
      >
        0 Comments
      </h2>
      <ul />
      <form
        className="c2"
        onSubmit={[Function]}
      >
        <h2>
          Submit a comment
        </h2>
        
        <div>
          <input
            className=""
            id="user"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
          <span
            className="message"
            data-testid="username-error-message"
          >
            Username cannot be empty
          </span>
        </div>
        <div>
          <textarea
            className=""
            id="content"
            name="comment"
            onChange={[Function]}
            placeholder="Write your comment here..."
            value=""
          />
          <span
            className="message"
            data-testid="comments-error-message"
          >
            Comment cannot be empty
          </span>
        </div>
        <button
          className="full"
          data-testid="submit"
          type="submit"
        >
          Submit
        </button>
      </form>
    </section>
    <footer>
      <button
        className="transparent go-back"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="button"
      >
        ‚Üê Back to the blog
      </button>
    </footer>
  </div>
</article>
`;

exports[`the <PostLayout /> component should render Post with comments 1`] = `
.c0 {
  padding: 0 1rem;
  max-width: 48rem;
  margin: 0 auto;
  text-align: center;
}

.c0 header {
  padding-top: 1.5rem;
}

.c0 header dl dd {
  line-height: 1.5rem;
  font-size: 1rem;
}

.c0 header h1 {
  line-height: 2.25rem;
  font-size: 1.875rem;
  font-weight: 800;
}

.c0 .post {
  padding-bottom: 2rem;
}

.c0 .post .author {
  padding: 2rem 0;
  border-bottom: 1px solid;
}

.c0 .post .author dd {
  font-size: 0.875rem;
  font-weight: 500;
  white-space: nowrap;
  line-height: 1.25rem;
}

.c0 .post .content {
  padding: 2rem 0;
  text-align: justify;
  border-bottom: 1px solid;
}

.c0 .post .content p {
  margin: 1.25em 0;
  font-size: 1rem;
  line-height: 1.75;
}

.c0 .post footer {
  text-align: justify;
  padding-top: 2rem;
}

.c0 .post footer .other-articles {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .post footer .other-articles .previous-article,
.c0 .post footer .other-articles .next-article {
  padding-bottom: 1rem;
}

.c0 .post footer .other-articles .previous-article h2,
.c0 .post footer .other-articles .next-article h2 {
  font-size: 0.75rem;
  line-height: 1rem;
  text-transform: uppercase;
}

.c0 .post footer .other-articles .previous-article button,
.c0 .post footer .other-articles .next-article button {
  padding-top: 0rem;
  text-transform: capitalize;
}

.c0 .post footer .go-back {
  padding-top: 1rem;
  float: right;
}

.c2 {
  margin: 2rem auto;
}

.c2 p {
  font-size: 1.25rem;
  line-height: 1.75rem;
}

.c2 div {
  margin: 2rem 0;
}

.c2 div input[type='text'],
.c2 div textarea {
  width: 100% !important;
  border-radius: 0.4rem;
  box-sizing: border-box;
  margin: 1rem 0;
  box-shadow: 0 0 20px 0 rgba(0,0,0,0.2),0 5px 5px 0 rgba(0,0,0,0.24);
  padding: 0.5rem;
  display: block;
}

.c2 div input[type='text']:focus-visible,
.c2 div textarea:focus-visible,
.c2 div input[type='text']:focus:not(:focus-visible),
.c2 div textarea:focus:not(:focus-visible) {
  outline: none;
  border: 2px solid;
}

.c2 div input[type='text'] {
  height: 2rem;
}

.c2 div textarea {
  height: 10rem !important;
  resize: none;
}

.c2 div .invalid {
  border: 2px solid !important;
}

.c2 div .message {
  font-size: 0.75rem;
  display: none;
}

.c2 div .show {
  display: block !important;
}

.c1 {
  margin: 2rem 0;
  text-align: justify;
}

.c1 ul {
  margin: 2rem 0;
}

.c1 ul li {
  margin: 1rem 0;
}

.c1 ul li h3 {
  font-size: 0.75rem;
}

.c1 ul li p {
  margin-top: 0.25rem;
  font-size: 0.5rem;
}

.c1 ul li .parent,
.c1 ul li .child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 ul li .child {
  width: calc(100% - 1.5rem);
  paddding: 2rem 0;
  margin: 1rem 0 1rem 1.5rem;
}

@media all and (min-width:960px) {
  .c2 {
    text-align: left;
  }

  .c2 div input[type='text'],
  .c2 div textarea {
    width: 70% !important;
  }
}

@media all and (min-width:960px) {
  .c1 ul li h3 {
    font-size: 0.8rem;
  }

  .c1 ul li p {
    margin-top: 0.25rem;
    font-size: 1rem;
  }

  .c1 ul li .child {
    max-width: 30rem;
    paddding: 2rem 0;
    margin: 1rem 0 1rem 3.5rem;
  }
}

<article
  className="c0"
>
  <header>
    <dl>
      <dt />
      <dd>
        <time
          aria-label="Published on Friday, January 1, 2021"
          dateTime="2021-01-01"
          tabIndex={0}
        >
          Friday, January 1, 2021
        </time>
      </dd>
    </dl>
    <h1>
      test
    </h1>
  </header>
  <div
    className="post"
  >
    <dl
      aria-label="Post written by author"
      className="author"
      tabIndex={0}
    >
      <dt />
      <dd>
        author
      </dd>
    </dl>
    <div
      className="content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "bla",
        }
      }
      tabIndex={0}
    />
    <section
      className="c1"
    >
      <h2
        tabIndex={0}
      >
        2 Comments
      </h2>
      <ul>
        <li>
          <div
            className="parent"
          >
            <div>
              <h3
                tabIndex={0}
              >
                test on 2021-06-28
              </h3>
              <p
                tabIndex={0}
              >
                banana
              </p>
            </div>
            <button
              className="full"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
            >
              Reply
            </button>
          </div>
          <div
            className="child"
          >
            <div>
              <h3
                role="text"
                tabIndex={0}
              >
                yolo response to test on 2021-06-28
              </h3>
              <p
                role="text"
                tabIndex={0}
              >
                bananan 2x
              </p>
            </div>
          </div>
        </li>
        <li />
      </ul>
      <form
        className="c2"
        onSubmit={[Function]}
      >
        <h2>
          Submit a comment
        </h2>
        
        <div>
          <input
            className=""
            id="user"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
          <span
            className="message"
            data-testid="username-error-message"
          >
            Username cannot be empty
          </span>
        </div>
        <div>
          <textarea
            className=""
            id="content"
            name="comment"
            onChange={[Function]}
            placeholder="Write your comment here..."
            value=""
          />
          <span
            className="message"
            data-testid="comments-error-message"
          >
            Comment cannot be empty
          </span>
        </div>
        <button
          className="full"
          data-testid="submit"
          type="submit"
        >
          Submit
        </button>
      </form>
    </section>
    <footer>
      <button
        className="transparent go-back"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="button"
      >
        ‚Üê Back to the blog
      </button>
    </footer>
  </div>
</article>
`;
